Helm Charts with base64 icon raise a pydantic error

If you have a `Chart.yaml` file containing a base64 png icon, pydantic raises : 

```
E       pydantic_core._pydantic_core.ValidationError: 1 validation error for Chart
E       metadata.icon
E         URL should have at most 2083 characters [type=url_too_long, input_value='data:image/png;base64,iV...RrulQAAAABJRU5ErkJggg==', input_type=str]
E           For further information visit https://errors.pydantic.dev/2.9/v/url_too_long
```

--- a/pyhelm3/models.py
+++ b/pyhelm3/models.py
@@ -12,6 +12,7 @@ from pydantic import (
     FilePath,
     PrivateAttr,
     TypeAdapter,
+    UrlConstraints,
     constr,
     field_validator,
 )
@@ -63,9 +64,14 @@ def validate_str_as(validate_type):
     return lambda v: str(adapter.validate_python(v))
 
 
+class PydanticDataUrl(PydanticAnyUrl):
+    _constraints = UrlConstraints(allowed_schemes=["data"])
+
+
 #: Annotated string types for URLs
 AnyUrl = t.Annotated[str, AfterValidator(validate_str_as(PydanticAnyUrl))]
 HttpUrl = t.Annotated[str, AfterValidator(validate_str_as(PydanticHttpUrl))]
+DataUrl = t.Annotated[str, AfterValidator(validate_str_as(PydanticDataUrl))]
 
 
 class ChartDependency(BaseModel):
@@ -149,7 +155,7 @@ class ChartMetadata(BaseModel):
     maintainers: t.List[ChartMaintainer] = Field(
         default_factory=list, description="List of maintainers for the chart."
     )
-    icon: t.Optional[HttpUrl] = Field(
+    icon: t.Optional[HttpUrl | DataUrl] = Field(
         None, description="URL to an SVG or PNG image to be used as an icon."
     )
     app_version: t.Optional[NonEmptyString] = Field(

--- a/pyhelm3/models.py
+++ b/pyhelm3/models.py
@@ -14,6 +14,7 @@ from pydantic import (
     FilePath,
     AnyUrl as PydanticAnyUrl,
     HttpUrl as PydanticHttpUrl,
+    UrlConstraints,
     constr,
     field_validator
 )
@@ -61,10 +62,13 @@ def validate_str_as(validate_type):
     adapter = TypeAdapter(validate_type)
     return lambda v: str(adapter.validate_python(v))
 
+class PydanticDataUrl(PydanticAnyUrl):
+    _constraints = UrlConstraints(allowed_schemes=["data"])
 
 #: Annotated string types for URLs
 AnyUrl = t.Annotated[str, AfterValidator(validate_str_as(PydanticAnyUrl))]
 HttpUrl = t.Annotated[str, AfterValidator(validate_str_as(PydanticHttpUrl))]
+DataUrl = t.Annotated[str, AfterValidator(validate_str_as(PydanticDataUrl))]
 
 
 class ChartDependency(BaseModel):
@@ -173,7 +177,7 @@ class ChartMetadata(BaseModel):
         default_factory = list,
         description = "List of maintainers for the chart."
     )
-    icon: t.Optional[HttpUrl] = Field(
+    icon: t.Optional[HttpUrl | DataUrl] = Field(
         None,
         description = "URL to an SVG or PNG image to be used as an icon."
     )

